controllers/notification/notification_controller.go

```diff
@@ -88,8 +88,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 
 	// if the environment is set to e2e do not attempt sending via mailgun
 	if config.Environment() != "e2e-tests" {
+		// get the notification environment
+		notificationTemplateSetName := config.Notifications().NotificationTemplateSetName()
 		// Send the notification via the configured delivery service
-		err = r.deliveryService.Send(notification)
+		err = r.deliveryService.Send(notification, notificationTemplateSetName)
```

Comment:
just a side note - it would be cool to separate the creation of templates & processing them from the actual sending over mailgun, so the first part 
https://github.com/codeready-toolchain/host-operator/blob/0ec0d89abee88a7527bd5a4648b6d3a979bcbf04/controllers/notification/mailgun_notification_delivery_service.go#L72-L108
could have been executed also as part of e2e tests and only the last part in the production
https://github.com/codeready-toolchain/host-operator/blob/0ec0d89abee88a7527bd5a4648b6d3a979bcbf04/controllers/notification/mailgun_notification_delivery_service.go#L110-L128
But let's do it as a separate story

Reply:
wonderful idea, created a new story - https://issues.redhat.com/browse/CRT-1736